src/main/java/org/xolstice/maven/plugin/protobuf/Protoc.java

```diff
@@ -652,7 +652,7 @@ public Builder addProtoFile(final File protoFile) {
             if (!protoFile.isFile()) {
                 throw new MojoConfigurationException("Proto file is not a file: " + protoFile.getAbsolutePath());
             }
-            if (!protoFile.getName().endsWith(".proto")) {
+            if (!protoFile.getName().matches(".*\\.proto\\d*$")) {
```

Comment:
This is not the only place where the `.proto` extension is being checked or filtered on. See for example `AbstractProtocMojo`.

Reply:
What's the best way to test my changes?

Reply:
There's a whole bunch of integration tests in the `src/it` directory.
In your case, the simplest approach will be to update tests 5-12. They are all pretty similar, they test different types of dependencies (reactor/non-reactor, main/test) and proto sources in both the default namespace and a custom namespace. You will need to drop simple messages with `proto2` and `proto3` extensions into source folders, and make the ones in `project2` depend on the ones in `project1`. Then update `verify.groovy` to check for the presence of the generated java files for all source protos.
That will give us a pretty good coverage of the changes and will be useful in the future as a part of regression test pack.